src/state/stake/useOwnerStakedPools.ts

```diff
@@ -27,5 +28,29 @@ export const useOwnerStakedPools = (farmSummaries: FarmSummary[]) => {
     return partition(farmSummaries, (farmSummary) => isStaked[farmSummary.stakingAddress])
   }, [farmSummaries, isStaked])
 
-  return { stakedFarms, unstakedFarms }
+  const uniqueUnstakedFarms = useMemo(() => unique(unstakedFarms), [unstakedFarms])
+
+  return { stakedFarms, unstakedFarms: uniqueUnstakedFarms }
+}
+
+function unique(farmSummaries: FarmSummary[]): FarmSummary[] {
+  const cache: Record<string, FarmSummary[]> = {}
+
+  const farmsGroupedByLp = farmSummaries.reduce((byLpAddress, farm) => {
+    const previous = byLpAddress[farm.lpAddress] || []
+
+    const likeFarms = [...previous, farm]
```

Comment:
I've been thinking lately about how a lot of React code gets spread often in the name of immutability. I think this could be more efficient and work the same if we just reused the same array or object

Reply:
could be, i find the main benefit of this spread is more a feeling of safety since mutable js methods differ in what is returned.